src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -114,17 +116,16 @@ public PathValidationResult validatePath(PathValidationData pathValidationData)
                 .flatMap(Set::stream)
                 .collect(Collectors.toSet());
 
-        validationsResult.addAll(executeValidations(InputData.of(data),
-                getPerPathValidations(data)));
+        validationsResult.addAll(executeValidations(InputData.of(data), getPerPathValidations(data)));
 
         GetPathsResult pceResult = validatePathComputation(data);
         String pceResponse = pceResult.isSuccess() ? "The path has been computed successfully" :
                 format("The path has not been computed successfully due to the following reasons: %s",
                         pceResult.getFailReasons().values());
 
         return PathValidationResult.builder()
-                .isValid(validationsResult.isEmpty() && pceResult.isSuccess())
-                .errors(new LinkedList<>(validationsResult))
+                .isValid(pceResult.isSuccess())
+                .validationMessages(new LinkedList<>(validationsResult))
```

Comment:
should we use addValidationMessages() method here and pass only the element(validationsResult) instead of  the current implementation? 
```suggestion
                .addValidationMessages(validationsResult)
```

Reply:
Actually, we can't do that. It works a bit differently. We add a `@Singular` annotation on a collection so that we have a shortcut for adding a single element to this collection. The method accepts the collection generic type. In this case `validationMessages` is a `List<String>`, so `addValidationMessage` accepts a String.
This is just a syntactic sugar for 
```java
    .validationMessages(Collections.singletonList("foo");
    .addValidationMessage("foo");
```